Sources/Node/Utilities/Errors.swift

```diff
@@ -1,55 +1,114 @@
 @_exported import Debugging
 
-public struct NodeError: Debuggable {
-    public let node: Node?
-    public let expectation: String
-    public let path: String
+public protocol NodeError: Debuggable {}
 
-    public let type: String = NodeError.unableToConvert // replacing the enum case to identify error type
-
-    internal init<S: SchemaWrapper>(node: S?, expectation: String, indexers: [PathIndexer] = []) {
-        self.node = node.flatMap { Node($0) }
-        self.expectation = expectation
-        self.path = indexers.path()
-    }
-
-    internal init(node: Node?, expectation: String, indexers: [PathIndexer] = []) {
-        self.node = node
-        self.expectation = expectation
-        self.path = indexers.path()
-    }
-}
-
-extension NodeError {
-    public static var unableToConvert: String { return "unableToConvert" }
+public enum Error: NodeError {
+    case invalidContainer(container: String, element: String)
+    case unableToConvert(input: Node?, expectation: String, path: [PathIndexer])
```

Comment:
I used `expectation: Any.Type` in the `ConfigError` and it worked nicely. Could be nice here as well.